python/main.py

```diff
@@ -110,81 +116,108 @@ async def get_image(image_name):
-                json.dump(DEFAULT_JSON_DATA, f, indent=2)
-
-        with open(JSON_DB, "r+", encoding="utf-8") as f:
-            content = f.read().strip()
-            data = json.loads(content) if content else DEFAULT_JSON_DATA
-            logger.info("Succeeded to open json file")
-
-            if "items" not in data:
-                data["items"] = []
-
-            existing_item = next((i for i in data["items"] if i["name"] == item.name), None)
-
-            if existing_item:
-                logger.info(f"Item already exists, updating image_name: {existing_item}")
-                existing_item["image_name"] = item["image_name"]
-            else:
-                new_item = item.dict()
-                data["items"].append(new_item)
-                logger.info(f"New item inserted: {new_item}")
-
-            f.seek(0)
-            json.dump(data, f, indent=2, ensure_ascii=False)
-            f.truncate()
+        cursor = db.cursor()
+        cursor.execute("SELECT id FROM categories WHERE name = ?", (item.category,))
+        category_row = cursor.fetchone()
+        
+        if category_row is None:
+            cursor.execute("INSERT INTO categories (name) VALUES (?)", (item.category,))
+            db.commit()  
+            category_id = cursor.lastrowid
+        else:
+            category_id = category_row["id"]
+
+        cursor.execute("SELECT id FROM items WHERE name = ?", (item.name,))
+        existing_item = cursor.fetchone()
+
+        if existing_item:
+            logger.info(f"Item already exists: {existing_item}")
+            raise HTTPException(status_code=400, detail="Item already exists")
+
+        cursor.execute(
+            "INSERT INTO items (name, category_id, image_name) VALUES (?, ?, ?)",
+            (item.name, category_id, item.image_name),
+        )
+        db.commit()
+        logger.info(f"New item inserted: {item.dict()}")
 
     except Exception as e:
+        db.rollback()
         logger.error(f"Failed to save item: {e}")
         raise HTTPException(status_code=500, detail="Failed to save item")
 
 
 
 @app.get("/items")
-def get_items():
+def get_items(db: sqlite3.Connection = Depends(get_db)):
     try:
-        if not JSON_DB.exists():
-            raise HTTPException(status_code=404, detail="Item not found")
-
-        with open(JSON_DB, "r", encoding="utf-8") as f:
-            content = f.read().strip()
-            data = json.loads(content) if content else DEFAULT_JSON_DATA
-
-        return {"items": data.get("items", [])}
-
+        cursor = db.cursor()
+        cursor.execute("""
+            SELECT items.id, items.name, categories.name AS category_name, items.image_name
+            FROM items
+            JOIN categories ON items.category_id = categories.id
+        """)
+        items = cursor.fetchall()
+
+        return {
+            "items": [
+                {"id": row["id"], "name": row["name"], "category": row["category"], "image_name": row["image_name"]}
```

Comment:
```suggestion
                {"id": row["id"], "name": row["name"], "category": row["category_name"], "image_name": row["image_name"]}
```

Maybe this should be this?